Packs/HPEArubaCentral/Integrations/HPEArubaCentralEventCollector/HPEArubaCentralEventCollector.py

```diff
@@ -0,0 +1,724 @@
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.user_name = user_name
+        self.user_password = user_password
+        self.customer_id = customer_id
+
+    def get_access_token(self, use_cached_token=True) -> str:
+        """
+         Get access token for Aruba Central API.
+         If one exists in the integration context and is not expired, returns it.
+         Otherwise, refreshes the access token using the refresh token and returns the new token.
+
+         Args:
+         use_cached_token (bool): Whether to use the cached access token if it exists and is not expired.
+                                  If set to false, the token will either be refreshed or a new one will be created.
+
+         Returns:
+             Valid access token to the Aruba Central API.
+        """
+        integration_context = get_integration_context()
+        access_token = integration_context.get('access_token')
+        expiry_time = integration_context.get('expiry_time', 0)
+        refresh_token = integration_context.get('refresh_token')
+
+        if use_cached_token and access_token and expiry_time > int(time.time()):
+            demisto.debug('Returning cached access token')
+            return access_token
+        elif isinstance(refresh_token, str):
+            demisto.debug('Refreshing access token.')
+            access_token, refresh_token, validity_duration = self.refresh_access_token(refresh_token)
+        else:
+            demisto.debug('Acquiring new access token via oauth.')
+            access_token, refresh_token, validity_duration = self.oauth_sequence()
+
+        integration_context.update({
+            'access_token': access_token,
+            'expiry_time': int(time.time()) + validity_duration,
+            'refresh_token': refresh_token
+        })
+        set_integration_context(integration_context)
+
+        return access_token
+
+    def refresh_access_token(self, refresh_token: str) -> tuple[str, str, int]:
+        """
+        Refreshes the access token using the provided refresh token.
+
+        Args:
+            refresh_token (str): Refresh token to be used
+
+        Returns:
+            access_token (str): The new access token.
+            refresh_token (str): The next refresh token.
+            expires_in (int): The validity duration of the new access token in seconds.
+        """
+        headers = {'Content-Type': 'application/json'}
+        params = {
+            'grant_type': 'refresh_token',
+            'client_id': self.client_id,
+            'client_secret': self.client_secret,
+            'refresh_token': refresh_token
+        }
+
+        try:
+            token_resp = self._http_request(
+                method='POST',
+                url_suffix='/oauth2/token',
+                headers=headers,
+                params=params,
+            )
+
+        except DemistoException as e:
+            if "Invalid refresh_token" in e.message:
```

Comment:
```suggestion
            if "Invalid refresh_token" in str(e):
```
Using str(e) ensures that the code works correctly in Python 3 and avoids potential issues with missing attributes.